Datagrid/ProxyQuery.php

```diff
@@ -48,6 +48,13 @@ class ProxyQuery implements ProxyQueryInterface
     protected $entityJoinAliases;
 
     /**
+     * The map of query hints.
+     *
+     * @var array
```

Comment:
Is it an associative array? How many dimensions? What does it contain? please describe.

Reply:
I added types in phpdoc and references to `\Doctrine\ORM\Query`. Is it good enough? We can pass one of [predefined hints](http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/dql-doctrine-query-language.html#query-hints) or define [custom](http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/cookbook/dql-custom-walkers.html#modify-the-output-walker-to-generate-vendor-specific-sql). It is associative array, same behaviour as in `Query`. The `name` is always a string, type of `value` depends on hint.